### PR TITLE
Enhance/add task

### PR DESCRIPTION
**Pull request by danieltolmon**
Enhance to let the user see, update, create and delete tasks.

Please ignore first commit (de3780a). it is revert afterwards.

### CODE REVIEW

File: src/app/models/taskCustom.ts
Comment (danielgek) [inline src/app/models/taskCustom.ts]: this needs to be changed

File: src/app/workflow-center/custom-task/custom-task.component.html
Comment (danielgek) [inline src/app/workflow-center/custom-task/custom-task.component.html]: this could be `clrModalSize="lg"`

File: src/app/workflow-center/custom-task/custom-task.component.html
Comment (danielgek) [inline src/app/workflow-center/custom-task/custom-task.component.html]: double quotes on 4

File: src/app/workflow-center/custom-task/custom-task.component.html
Comment (danielgek) [inline src/app/workflow-center/custom-task/custom-task.component.html]: also here

File: src/app/workflow-center/custom-task/custom-task.component.html
Comment (danielgek) [inline src/app/workflow-center/custom-task/custom-task.component.html]: same as below

File: src/app/workflow-center/custom-task/custom-task.component.html
Comment (danielgek) [inline src/app/workflow-center/custom-task/custom-task.component.html]: space after task

File: src/app/workflow-center/custom-task/custom-task.component.html
Comment (danielgek) [inline src/app/workflow-center/custom-task/custom-task.component.html]: same as below

File: src/app/workflow-center/custom-task/custom-task.component.html
Comment (danielgek) [inline src/app/workflow-center/custom-task/custom-task.component.html]: hummm this comparison is strange

File: src/app/workflow-center/custom-task/custom-task.component.html
Comment (danielgek) [inline src/app/workflow-center/custom-task/custom-task.component.html]: name don't need to be binded

File: src/app/workflow-center/custom-task/custom-task.component.html
Comment (danielgek) [inline src/app/workflow-center/custom-task/custom-task.component.html]: double quote

**Review by danielgek — CHANGES_REQUESTED** [review]

File: src/app/models/taskCustom.ts
Comment (danieltolmon) [reply src/app/models/taskCustom.ts]: it might be a JSON

File: src/app/workflow-center/custom-task/custom-task.component.ts
Comment (danieltolmon) [inline src/app/workflow-center/custom-task/custom-task.component.ts]: this one

File: src/app/workflow-center/custom-task/custom-task.component.ts
Comment (danieltolmon) [inline src/app/workflow-center/custom-task/custom-task.component.ts]: no task, it should be properties

File: src/app/workflow-center/custom-task/custom-task.component.html
Comment (danieltolmon) [reply src/app/workflow-center/custom-task/custom-task.component.html]: The options (and properties) can arribe as empty json. In this case, the link should not be shown since they don't have anything to show. I changed it to only `(task.options | json) != '{}'` . If i leve it as only `task.options`, it will always show the link, since the options is always defined

File: src/app/workflow-center/custom-task/custom-task.component.html
Comment (danieltolmon) [reply src/app/workflow-center/custom-task/custom-task.component.html]: all this mistakes comes from rackHD project. Maybe we should do a refactor to clean all of them (they are quite a lot). Should be open an issue?

**Review by danielgek — APPROVED** [review]

**Comment by danielgek** [comment]
LGTM 🔥 if feature wise is complete, can be merged

**Comment by danieltolmon** [comment]
ahh wait, I got a question for @jsaguiar...
As we commented, I add the feature of custom tasks inside the workflow center:

Screenshot 2020-01-21 at 11 06 35

But I think it has more sense to add it inside the management center. Since you are managing the tasks (you can update, create and delete them). It has also the same format from the others: a table where you can see all tasks with buttons to select, delete and create.

Screenshot 2020-01-21 at 11 06 46

Do I add tasks inside the management center instead of having it in the workflow center?

Also, when I update and create... I call the same http PUT request. It's what RackHD does with all of them (when you create/update a workflow, node etc etc... it calls the same endpoint) before merge it, I want to ensure if this it the way. If the backend has the logic of knowing if the task/workflow/node...  exists to update it, or if not, it creates a new one.

**Comment by jsaguiar** [comment]
UI, I agree that you  should put it in workflow center.
Regarding API, looking from what we have now, it seems that it works like that. But feel free to test to add / update and remove **NEW** tasks in the environment you have access.

File: package.json
Comment (danielgek) [inline package.json]: @danieltolmon i would prefer ti have this later on in a new branch ;)

File: src/app/management-center/workflows/workflows.component.ts
Comment (danielgek) [inline src/app/management-center/workflows/workflows.component.ts]: i guess you can remove this

File: package.json
Comment (danieltolmon) [reply package.json]: Okai, I create a branch to add it

**Comment by danielgek** [comment]
@danieltolmon this one is also good to go

File: src/app/management-center/tasks/tasks.component.html
Comment (danieltolmon) [inline src/app/management-center/tasks/tasks.component.html]: comment!!  👀

Comment (danieltolmon) [reply src/app/management-center/tasks/tasks.component.html]: I do it myself

**Pull request merged**
